horovod/common/common.cc

```diff
@@ -137,11 +139,67 @@ void server_affinity_set(int affinity) {
   }
 }
 #else
-void server_affinity_set(int affinity) {
+void set_affinity(int affinity) {
   // TODO(travis): explore equivalent for macOS
   throw std::runtime_error("Environment variable HOROVOD_THREAD_AFFINITY is not supported on macOS.");
 }
 #endif
 
+void parse_and_set_affinity(const char* affinity, int local_size, int local_rank) {
+  if (affinity == nullptr) {
+    return;
+  }
+
+  size_t affinity_len = strlen(affinity);
+
+  // copy is needed because strsep is going to modify the buffer
+  char* affinity_copy = (char*)calloc(affinity_len + 1, sizeof(char));
+  memcpy(affinity_copy, affinity, affinity_len);
+  char* tmp = affinity_copy;
+  char *endptr;
+
+  std::vector<int> core_ids(local_size);
+  int count = 0;
+
+  while (*tmp != 0 && count < local_size) {
+    auto core_id_str = strsep(&tmp, ",");
+    errno = 0;
```

Comment:
It seems `errno` is set to zero here and never updated?

Reply:
errno value will be changed by strtol function in case something bad happens